### ISSUE TITLE
add current server

### ISSUE THREAD

**Issue by MaximilianKohler** [ValveSoftware/halflife#508]
This button doesn't do anything for me.

win7 x64

**Comment by SamVanheer** [comment]
This still happens.

I did a little research and i found that the problem appears to be a missing byte swap operation in `CServerBrowserDialog::OnConnectToGame`. The server browser stores off the connection data here, and adds it to the history list as well.

The data that is stored is taken directly from the given KeyValues data, but the IP address that gets passed to `ISteamMatchmaking::AddFavoriteGame` is byte swapped. If the IP address is invalid it might not be added to the list of favorites to begin with.

I did test to see if the button was not being initialized properly, but i ruled it out. `Half-Life/platform/servers/InternetGamesPage.res` contains the command used for the button, changing it to `AddServerByName` works.

I did notice that `AddServerByName` has both its own command handler and a case in `OnCommand` while `AddCurrentServer` is only handled in `OnCommand`, so if the byte swap thing isn't the cause of this, perhaps this is why it isn't working. Based on what i know about VGUI2 that shouldn't be the case though.
